File: python/b/filequestion.py
import b.internal as b
from b.question import Question

class FileQuestion(Question):
    def __init__(self, filename):
        super(FileQuestion, self).__init__(
            ptr=b.lib.b_file_question_allocate(filename),
            vtable=b.lib.b_file_question_vtable())

File: haskell/B/Foreign/FileQuestion.hs
{-# LANGUAGE ForeignFunctionInterface #-}

module B.Foreign.FileQuestion where

import Foreign.C
import Foreign.Ptr

import B.Foreign.Question (AnyQuestion, QuestionVTable)

foreign import ccall "b_file_question_allocate"
  allocate
    :: CString
    -> IO (Ptr AnyQuestion)

foreign import ccall "b_file_question_deallocate"
  deallocate
    :: Ptr AnyQuestion
    -> IO ()

foreign import ccall "b_file_question_file_path"
  filePath
    :: Ptr AnyQuestion
    -> IO CString

foreign import ccall "b_file_question_vtable"
  vtable
    :: IO (Ptr QuestionVTable)
